src/mongo/db/query/datetime/date_time_support.cpp

```diff
@@ -122,6 +131,16 @@ TimeZone TimeZoneDatabase::utcZone() {
     return TimeZone{nullptr};
 }
 
+Date_t TimeZoneDatabase::fromString(StringData dateString) const {
+	timelib_time *t;
+	timelib_error_container *errors;
+
+	t = timelib_strtotime((char*) dateString.toString().c_str(), dateString.size(), &errors, _timeZoneDatabase, TimeZoneDatabase::getTZInfoWrapper);
```

Comment:
The reference to getTZInfoWrapper fails (expectedly) to compile:

```
src/mongo/db/query/datetime/date_time_support.cpp: In member function 'mongo::Date_t mongo::TimeZoneDatabase::fromString(mongo::StringData) const':
src/mongo/db/query/datetime/date_time_support.cpp:138:144: error: invalid use of non-static member function 'timelib_tzinfo* mongo::TimeZoneDatabase::getTZInfoWrapper(char*, const _timelib_tzdb*, int*)'
  t = timelib_strtotime((char*) dateString.toString().c_str(), dateString.size(), &errors, _timeZoneDatabase, TimeZoneDatabase::getTZInfoWrapper);
                                                                                                                                                ^
src/mongo/db/query/datetime/date_time_support.cpp:80:17: note: declared here
 timelib_tzinfo *TimeZoneDatabase::getTZInfoWrapper(char *tz_id, const _timelib_tzdb *db, int *error)
                 ^~~~~~~~~~~~~~~~
```

```timelib_strtotime``` needs access to the timelib_tzdb database as strings can also contain timezones. The wrapper is a static method that needs to be called in such a way it can also use ```_timeZones``` as is shown at https://github.com/mongodb/mongo/pull/1156/files#diff-95abc574b79f05301d8807add70d6961R80

I don't know how to solve this.